gamemodes/terrortown/entities/weapons/weapon_zm_carry.lua

```diff
@@ -359,116 +334,140 @@ end
 function SWEP:AllowPickup(target)
 	local phys = target:GetPhysicsObject()
 	local ply = self:GetOwner()
+	local ctype = DetermineCarryType(target)
 
 	return IsValid(phys) and IsValid(ply)
 		and not phys:HasGameFlag(FVPHYSICS_NO_PLAYER_PICKUP)
 		and phys:GetMass() < CARRY_WEIGHT_LIMIT
 		and not PlayerStandsOn(target)
 		and target.CanPickup ~= false
-		and (target:GetClass() ~= "prop_ragdoll" or GetGlobalBool("ttt_ragdoll_carrying"))
-		and (not target:IsWeapon() or GetGlobalBool("ttt_weapon_carrying"))
+		and (ctype ~= CARRY_TYPE_RAGDOLL or cvAllowRagCarry:GetBool())
+		and (ctype ~= CARRY_TYPE_WEAPON or cvAllowWepCarry:GetBool())
 		---
 		-- @realm shared
 		and not hook.Run("TTT2PlayerPreventPickupEnt", ply, target)
 end
 
+---
+-- @param Entity target
+-- @realm shared
+function SWEP:AllowPush(target)
```

Comment:
Rename to something like `SWEP:CanBeMoved(target)`  because right now this sounds like a check function only for the push / pull mechanic, but it isn't.